spec/spec_helper.rb

```diff
@@ -52,5 +52,20 @@
     if ex.metadata[:js]
       Capybara.reset!
     end
+
+    # Allow Sentry to pick up messages triggered in our CI test build (and not locally).
+    # Sends a warning when tests are retried so we can track indeterminate tests.
+    if ENV.key?("SENTRY_DSN") && ex.metadata[:retry_exceptions]
+      # Disable WebMock so we can send events to Sentry
+      # Add sleep to avoid race condition
+      WebMock.disable!
+      sleep(3.seconds)
```

Comment:
I've introduced some sleep methods between WebMock turn on/off. This is because I was experiencing a race condition where the events were not sending to Sentry, presumably because WebMock was not in the right state. Adding the sleeps has fixed this although might not be the best way of tackling it.

Reply:
Yeah, we should definitely come back to this.